netx/httptransport/http3transport.go

```diff
@@ -7,27 +7,18 @@ import (
 
 	"github.com/lucas-clemente/quic-go"
 	"github.com/lucas-clemente/quic-go/http3"
+	"github.com/ooni/probe-engine/netx/dialer"
 )
 
-// HTTP3Dialer is the definition of dialer for HTTP3 transport assumed by this package.
-type HTTP3Dialer interface {
-	Dial(network, addr string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error)
-}
-
-// HTTP3ContextDialer is a dialer for HTTP3 transport using Context.
-type HTTP3ContextDialer interface {
-	DialContext(ctx context.Context, network, addr string, tlsCfg *tls.Config, cfg *quic.Config) (quic.EarlySession, error)
-}
-
-// HTTP3WrapperDialer is a HTTP3Dialer that wraps a ContextDialer
+// QUICWrapperDialer is a QUICDialer that wraps a ContextDialer
```

Comment:
(Maybe we can check whether perhaps they implemented this functionality?)

Reply:
In the quic-go library, they only allow one function signature to be passed to the RoundTripper for Dialing, which is the one implemented by `QUICWrapperDialer` in this PR. This signature does not allow passing a context, which is why I did not consider it suitable to use for all of the QUIC dialers. The `QUICContextDialer` uses context.
And now that I'm looking into it, I realize that I don't even use the `QUICBaseDialer` anymore. :sweat_smile: Ok, so now we're down to 2 QUIC dialers.

Reply:
Too bad there are no party parrots on GitHub! 🎉

Reply:
IKR? :D